shell/platform/windows/flutter_windows_engine_unittests.cc

```diff
@@ -223,6 +243,44 @@ TEST_F(FlutterWindowsEngineTest, RunWithoutANGLEUsesSoftware) {
   modifier.embedder_api().Shutdown = [](auto engine) { return kSuccess; };
 }
 
+TEST_F(FlutterWindowsEngineTest, RunWithoutANGLEOnImpellerFailsToStart) {
+  SetEnvironmentVariable("FLUTTER_ENGINE_SWITCHES", "1");
+  SetEnvironmentVariable("FLUTTER_ENGINE_SWITCH_1", "enable-impeller=true");
```

Comment:
Hm, I believe this test runs in parallel with the `FlutterProjectBundle` tests. For example:

https://github.com/flutter/engine/blob/1b44d7ee1a609cb9743940431704f5f1a50653e5/shell/platform/windows/flutter_project_bundle_unittests.cc#L69-L78

If so, we may run into concurrency issues. It looks like `FlutterProjectBundle` supported overriding the switches using `SetSwitches`, but that's currently broken:

https://github.com/flutter/engine/blob/3f50b7d51678a8899ee390ea22c0ba7a03bcdbef/shell/platform/windows/flutter_project_bundle.cc#L85-L92

Would you mind fixing that `SetSwitches` method and then moving this test to `SetSwitches`? Apologies for the mess. Feel free to skip this if the concurrency concern is invalid. Or also feel free to slap a TODO and I can fix it up in a follow-up change.

Reply:
Yeah I noticed that but I wasn't sure if this was the correct way to do it or not. If that is supposed to work then happy to fix it, seems like a better way to configure test state..